### PR TITLE
WishList & Cart Functionality

### PR DESCRIPTION
**Pull request by ShekharDhangar**
Note: The deployed link is not working due to .env file issue.But the code works in local machine. It will soon work on production.
I have added Wishlist and Cart Functionality. 
Note: The Branch is different due to mistake.
Please check below files.
Note: Ignore Toast Related Code.  Can ignore Styles as mostly are same as previous ecomm project. Also, please ignore empty cart page and wishlist Page.
Few files have short codes.
- [ ] (src/App.js)
- [ ] (src/Reducer/productReducer.js)
- [ ] (src/components/CartProductCard/CartProductCard.jsx)
- [ ] (src/components/Navbar/Navbar.jsx)
- [ ] (src/components/ProductCard/ProductCard.jsx)
- [ ] (src/context/cartContext.js)
- [ ] (src/context/productContext.js)
- [ ] (src/context/wishlistContext.js)
- [ ] (src/index.js)
- [ ] (src/pages/AuthenticationPage/AuthenticationComp/AuthLogin/AuthLogin.jsx)
- [ ] (src/pages/AuthenticationPage/AuthenticationComp/AuthSignUp/AuthSignUp.jsx)
- [ ] (src/pages/AuthenticationPage/AuthenticationComp/ForgotPassword/ForgotPassword.jsx)
- [ ] (src/pages/Cart/Cart.jsx)
- [ ] (src/pages/Wishlist/Wishlist.jsx)

### CODE REVIEW

**Comment by mikesch803** [comment]
hey shekhar, your deployed changes are not working, please check

File: src/App.js [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
     const InitialProductsData = await getProductsData();
     if (InitialProductsData) {
       dispatch({ type: "INITIAL PRODUCTS", payload: InitialProductsData })
+      dispatch({ type: "LOADER", payload: false });
     }
+    const encodedToken = localStorage.getItem("token");
+    dispatch({ type: "USERSIGNED", payload: encodedToken })
   }
   useEffect(() => {
     loadProductsData();
```
Comment (mikesch803) [inline src/App.js]: suggestion: this is also fine but here you can use iife in useEffect itself instead normal function

File: src/Reducer/productReducer.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
-import { contextInitialState } from "../context/context";
-
 function productReducer(state, { type, payload }) {
```
Comment (mikesch803) [inline src/Reducer/productReducer.js:1]: Its great you destructed already and one thing try to use fat arrow function

File: src/components/navItems.js [hunk 3]
```diff
@@ -23,24 +23,27 @@ const navItems = [
 
 const navIconsData = [
```
Comment (mikesch803) [inline src/components/navItems.js:24]: This is nice, you have data for links

File: src/context/cartContext.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    return (
+        <CartContext.Provider value={{ Cart, setCart, addToCart, removeFromCart, changeCartQuantity }} >
+            {children}
+        </CartContext.Provider>
+    )
+}
+
+const useCartContext = () => useContext(CartContext);
```
Comment (mikesch803) [inline src/context/cartContext.js:70]: useCart is fine for naming

**Review by mikesch803 — APPROVED** [review]
Hey I put some suggestion, One thing  is try create small pr as this pr has wishlist,cart, and even authentication ,else you code is good and understandable

File: src/context/cartContext.js (hunk 4 above)
Comment (ShekharDhangar) [reply src/context/cartContext.js:70]: Its just for getting know that is a context

File: src/App.js (hunk 1 above)
Comment (ShekharDhangar) [reply src/App.js]: done

File: src/components/CartProductCard/CartProductCard.jsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
+import { useCartContext, useWishlistContext } from "../../context/context";
+import "./CartProductCard.css";
+function CartProductCard({ productCard }) {
+  const { removeFromCart, changeCartQuantity } = useCartContext();
+  const { Wishlist, addToWishlist } = useWishlistContext();
+  const [moveLoader, setMoveLoader] = useState(false);
+
+  function isPresentInState(data, state) {
```
Comment (yash9263) [inline src/components/CartProductCard/CartProductCard.jsx:9]: extract these functions into a utility `utils` folder these can be used at different places

File: src/components/ProductCard/ProductCard.jsx [hunk 6]
```diff
@@ -1,18 +1,67 @@
...
+  
   const ratingsArray = [];
   for (let i = 1; i <= productCardDetails.starRating; i++) {
     ratingsArray.push({
       icon: <AiFillStar key={i} className="size-xs icon" />,
     });
   }
 
+
+  function isPresentInState(data, state) {
```
Comment (yash9263) [inline src/components/ProductCard/ProductCard.jsx:23]: the same function you can import here to use if you extract it

**Review by yash9263 — APPROVED** [review]

**Comment by yash9263** [comment]
You can follow through the comments made above else everything looks good.

File: src/components/ProductCard/ProductCard.jsx (hunk 6 above)
Comment (ShekharDhangar) [reply src/components/ProductCard/ProductCard.jsx:23]: Okay Made the changes in next Pr.

File: src/components/CartProductCard/CartProductCard.jsx (hunk 5 above)
Comment (ShekharDhangar) [reply src/components/CartProductCard/CartProductCard.jsx:9]: Okay Made the changes in next Pr.

**Pull request merged**